backend/cmd/headlamp.go

```diff
@@ -1047,10 +1051,174 @@ func handleClusterHelm(c *HeadlampConfig, router *mux.Router) {
 	})
 }
 
-func handleClusterAPI(c *HeadlampConfig, router *mux.Router) {
+func readFileContents(filePath string) []byte {
+	file, err := os.Open(filePath)
+	if err != nil {
+		log.Printf("Error opening file: %s", err)
+		return nil
+	}
+	defer file.Close()
+
+	var content []byte
+
+	scanner := bufio.NewScanner(file)
+
+	for scanner.Scan() {
+		content = append(content, scanner.Bytes()...)
+	}
+
+	if err := scanner.Err(); err != nil {
+		log.Printf("Error reading file: %s", err)
+		return nil
+	}
+
+	return content
+}
+
+func handleClusterAPI(c *HeadlampConfig, router *mux.Router) { //nolint:funlen,gocognit
 	router.PathPrefix("/clusters/{clusterName}/{api:.*}").HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		clusterName := mux.Vars(r)["clusterName"]
 
+		// checking if cluster exists, if not check if the request headers as cluster information
+		kubeconfig := r.Header.Get("KUBECONFIG")
+		if kubeconfig != "" { //nolint:nestif
+			decodedKubeconfig, err := base64.StdEncoding.DecodeString(kubeconfig)
+			if err != nil {
+				log.Printf("Error: deconding decodedKubeconfig: %s", err)
+				http.NotFound(w, r)
+				return
+			}
+
+			var kubeconfig Kubeconfig
+			err = yaml.Unmarshal(decodedKubeconfig, &kubeconfig)
+			if err != nil {
+				log.Printf("Error: unmarshaling kubeconfig: %s", err)
+				http.NotFound(w, r)
+				return
+			}
+
+			var (
+				serverURL                string
+				certificateAuthorityData []byte
+				clientCertificateData    []byte
+				clientKeyData            []byte
+				clusterFound, userFound  bool
+			)
+
+			for _, cluster := range kubeconfig.Clusters {
+				if cluster.Name == clusterName {
+					clusterName = cluster.Name
+					serverURL = cluster.Cluster.Server
+					if cluster.Cluster.CertificateAuthority != "" {
+						certificateAuthorityData = readFileContents(cluster.Cluster.CertificateAuthority)
+					} else {
+						certificateAuthorityData, err = base64.StdEncoding.DecodeString(cluster.Cluster.CertificateAuthorityData)
+						if err != nil {
+							log.Printf("Error: deconding certificateAuthorityData: %s", err)
+							http.NotFound(w, r)
+							return
+						}
+					}
+					clusterFound = true
+					break
+				}
+			}
+
+			for _, user := range kubeconfig.Users {
+				if user.Name == clusterName {
+					if user.User.ClientCertificate != "" {
+						clientCertificateData = readFileContents(user.User.ClientCertificate)
```

Comment:
You can use the [clientcmd.Load](https://pkg.go.dev/k8s.io/client-go/tools/clientcmd#Load) function to load the apiConfig from the kubeconfig file contents.